### ISSUE TITLE
Module.* is not a function

### ISSUE THREAD

**Issue by isolin** [alfg/ffprobe-wasm#13]
Hi! I just tried to reproduce your example webpage and get all up and running, but for all day I am struggling with the following: both `Module.get_file_info` as well as `Module.get_frames` throw errors: `Module.get_file_info is not a function` and `Module.get_frames is not a function`.

I believe that the COEP and COOP headers are set correctly for my test server (before I was getting other errors). The entry script `ffprobe-worker.js` receives the command messages and runs until line 17 and 51, respectively.

I do not use the node server, but host to files statically. I also switched from Vue to a very simple vanilla js. The debugger of Firefox shows that all files are fetched successfully, including the wasm file. I am not aware that I left something out, but one never knows. Your hosted example page runs perfect for me, so it won't be a browser issue.

Do you please have any idea what could be causing the issues or how I could get more information?

**Comment by alfg** [comment]
Hi @isolin, can you verify the COEP/COOP headers are in the response? You should see them in if you inspect via developer tools:
![Capture](https://user-images.githubusercontent.com/702541/147864473-ae683339-37a4-4f3b-8e8c-a30bb1a9a935.PNG)

I tried rebuilding the project from scratch just to be sure and it still works locally.

I also have a more simplified example without Vue if you haven't checked it out yet:
https://github.com/alfg/ffmpeg-webassembly-example

**Comment by alfg** [comment]
Since you're hosting the files statically, also make sure the `Content-Type: application/wasm` mimetype is properly set, in addition to the COEP/COOP headers. I ran into this issue before when hosting via NGINX.

**Comment by isolin** [comment]
Thank you for the really quick response! After I posted the question, I already went to bed :)

Both settings are correct, I suppose it must be something else.
![Screenshot at 2022-01-02 12-16-29](https://user-images.githubusercontent.com/9684905/147874142-78307791-7d96-4551-93e7-fa495e4c3929.png)

Thank you for the hint to the simpler example. I will try it out, but I suppose I will run into the same issue. My goal is to obtain the duration, frame count and frames timestamps of a video so that I can accurately process it further by using opencv.js.

**Comment by isolin** [comment]
@alfg I am happy to report that it works now! I noticed [this SO question](https://stackoverflow.com/questions/49516828/emscripten-wasm-in-web-worker-module-is-not-an-object-or-function) that mentioned that the worker referencing a wasm script needed to be in the main file to run and that was my problem as well. I noticed that you do a similar assignment in the Vue root. I originally created the worker in `app.js`, but now when I moved the `new Worker` call to the html page as an inlined script it finally started to work. Thank you once again for the quick response and awesome script.
